### PR TITLE
docs(dependency injection): add revamped guide focused on standalone

### PR DESCRIPTION
**Pull request by MarkTechson**
* Updates two of the md files to prioritize standalone components/architecture
* Updates three examples referenced in the guides
* Updates the svg diagram to reference the EnvironmentInjector instead of the ModuleInjector

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Currently the docs are focused on NgModules.

Issue Number: N/A

## What is the new behavior?
The dependency injection guide prioritizes standalone architecture

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
Please take care to read the updated `.md` documents to ensure accuracy. The shift from `ModuleInjector` to `EnvironmentInjector` in the writing can be a bit tricky and easy to conflate. Thank you!

### CODE REVIEW

File: aio/content/guide/dependency-injection.md [hunk 1]
```diff
@@ -34,18 +35,28 @@ class HeroListComponent {}
 
 When you register a provider at the component level, you get a new instance of the service with each new instance of that component.
 
-* At the NgModule level, using the `providers` field of the `@NgModule` decorator. In this scenario, the `HeroService` is available to all components, directives, and pipes declared in this NgModule or other NgModule which is within the same ModuleInjector applicable for this NgModule. When you register a provider with a specific NgModule, the same instance of a service is available to all applicable components, directives and pipes.
+
+* At the application level, using the `providers` field of the `AppConfg` object passed to the `bootstrapApplication` function. In this scenario, the `HeroService` is available to all components, directives, and pipes declared in this NgModule or other NgModule which is within the same ModuleInjector applicable for this NgModule. When you register a provider with a specific NgModule, the same instance of a service is available to all applicable components, directives and pipes.
```
Comment (jelbourn) [inline aio/content/guide/dependency-injection.md:39]: I think this first sentence is incomplete?

Comment (jelbourn) [inline aio/content/guide/dependency-injection.md:39]: Is the intent to not talk about `NgModule` in this paragraph?

File: aio/content/images/guide/dependency-injection/injectors.svg
Comment (jelbourn) [inline aio/content/images/guide/dependency-injection/injectors.svg:1]: Can we make a diagram with mermaid for this and save the text representation for when we want to change it later?

**Review by twerske — APPROVED** [review]

File: aio/content/guide/dependency-injection.md (hunk 1 above)
Comment (MarkTechson) [reply aio/content/guide/dependency-injection.md:39]: Lol true - it was in complete in the past, too but now we shall change the course of history.

Comment (MarkTechson) [reply aio/content/guide/dependency-injection.md:39]: Good point! Updated

**Review by bencodezen — APPROVED** [review]
LGTM!

**Review by jelbourn — APPROVED** [review]
LGTM

**Review by pkozlowski-opensource — APPROVED** [review]

**Comment by pkozlowski-opensource** [comment]
This PR was merged into the repository by commit 63b8108f66f19e059e8d31af8720b3ad1c4e92ad.

**Pull request closed without merging**